hack/lib/log/system.sh

```diff
@@ -167,22 +167,24 @@ function os::log::system::internal::plot() {
         local header
         header="${headers[$i]}"
 
-        if (( $i == ${#headers[@]} - 1 )); then
+        if (( i == ${#headers[@]} - 1 )); then
             # we need x-tick labels on the bottom plot
             gnuplot_directive+=( "set xtics format '%H:%M:%S' rotate by -90" )
         else
             gnuplot_directive+=( "set format x ''" )
         fi
 
-        gnuplot_directive+=( "plot \"${datafile}\" using 1:$(( $i + 1 )) title \"${header}\" with lines" )
+        gnuplot_directive+=( "plot \"${datafile}\" using 1:$(( i + 1 )) title \"${header}\" with lines" )
     done
 
     # concatenate the array with newlines to get the final directive to send to gnuplot
     gnuplot_directive="$( IFS=$'\n'; echo "${gnuplot_directive[*]}" )"
 
-    echo -e "\$ gnuplot <<< ${gnuplot_directive}\n" >> "${LOG_DIR}/gnuplot.log"
-    gnuplot <<< "${gnuplot_directive}" >> "${LOG_DIR}/gnuplot.log" 2>&1
-    echo -e "\n\n" >> "${LOG_DIR}/gnuplot.log"
+    {
+        echo -e "\$ gnuplot <<< ${gnuplot_directive}\n"
+        gnuplot <<< "${gnuplot_directive}" 2>&1
+        echo -e "\n\n"
```

Comment:
I am a fan of `printf`:
```bash
        printf '$ gnuplot <<< %s\n' "${gnuplot_directive}"
        gnuplot <<< "${gnuplot_directive}" 2>&1
        printf '\n\n'
```
Advantages:
• Because I am single-quoting the format string, I do not need to worry about escaping literals such as the `$` in the format string or know about Bash's weird semantics around `\n`.
• Because I am using `%s` to let `printf` interpolate `${gnuplot_directive}` into the string, I do not need to think about whether `${gnuplot_directive}` might have escape sequences that `echo -e` might interpret.
• It's POSIX! (FWIW.)